File: Problem_1.java
// 1. Two Sum (EASY) (SOLVED)
// https://leetcode.com/submissions/detail/1804693191/

import java.util.*;

class Problem_1 {
    public int[] twoSum(int[] nums, int target) {

    Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[]{map.get(complement), i};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: Problem_1.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # HashMap of complements
        output = []
        
        complements = {}

        for i, num in enumerate(nums):
            if num in complements:
                return [complements[num], i]
            complement = target - num
            complements[complement] = i
        return []
            

if __name__ == "__main__":
    mySolution = Solution()
    print(mySolution.twoSum([-1,-2,-3,-4,-5], -8))